src/Nox.ClientApi.Tests/Tests/Controllers/CountriesControllerTests.cs

```diff
@@ -1390,13 +1390,89 @@ public async Task Patch_Number_ShouldUpdateNumberOnly()
             var postResult = await PostAsync<CountryCreateDto, CountryDto>(Endpoints.CountriesUrl, createDto);
             var headers = CreateEtagHeader(postResult!.Etag);
 
-            var patchResult = await PatchAsync<CountryDto, CountryDto>($"{Endpoints.CountriesUrl}/{postResult!.Id}", updateDto, headers);
+            var patchResult = await PatchAsync<CountryUpdateDto, CountryDto>($"{Endpoints.CountriesUrl}/{postResult!.Id}", updateDto, headers);
 
             //Assert
             patchResult!.Population.Should().Be(expectedNumber);
             patchResult!.Name.Should().Be(expectedName);
         }
 
+        [Fact]
+        public async Task Patch_RequiredNameIsMissing_ShouldUpdatePopulation()
+        {
+            //This test is to verify the case that if the required Name is missing in delta Patch request, it should still update other fields and do not throw BadRequest
+            
+            // Arrange
+            var expectedPopulation = _fixture.Create<int>();
+            var countryResponse = await PostAsync<CountryCreateDto, CountryDto>(Endpoints.CountriesUrl,
+                new CountryCreateDto
+                {
+                    Name = _fixture.Create<string>(),
+                    Population = _fixture.Create<int>()
+                });
+
+            // Act            
+            var headers = CreateEtagHeader(countryResponse!.Etag);
+            var patchResponse = await PatchAsync<CountryUpdateDto, CountryDto>(
+                $"{Endpoints.CountriesUrl}/{countryResponse!.Id}",
+                new CountryUpdateDto
+                {
+                    Population = expectedPopulation
+                }, 
+                headers);
+
+            // Assert
+            patchResponse!.Population.Should().Be(expectedPopulation);
+            patchResponse!.Name.Should().Be(countryResponse!.Name);
+        }
+
+        [Fact]
+        public async Task Patch_RequiredName_IsSetToNull_ShouldFail()
+        {
+            // Arrange
+            var countryResponse = await PostAsync<CountryCreateDto, CountryDto>(Endpoints.CountriesUrl,
+                new CountryCreateDto
+                {
+                    Name = _fixture.Create<string>()
+                });
+
+            // Act            
+            var headers = CreateEtagHeader(countryResponse!.Etag);
+            var patchResponse = await PatchAsync(
+                $"{Endpoints.CountriesUrl}/{countryResponse!.Id}",
+                new CountryUpdateDto { },
+                headers,
+                false);
+
+            // Assert
+            patchResponse!.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+        }
+
+        [Fact(Skip = "Patch should update Compound nox types")]
+        public async Task Patch_CompoundNoxTypes_ShouldUpdate()
```

Comment:
@rochar I've found a new issue with deserializing CompoundNoxTypes and created a bug for it. Please see NOX-254 for more details and prioritize it accordingly.

Reply:
main is green again, if you rebase this branch will start pass the build